classes/Cart.php

```diff
@@ -92,6 +92,7 @@ class CartCore extends ObjectModel
     protected static $_carriers = null;
     protected static $_taxes_rate = null;
     protected static $_attributesLists = array();
+    protected static $_total_shipping = array();
```

Comment:
I think that for memoized vars, it is a way better pratice to just declare `static $totalShipping;` at the beginning of the public method.

It is better for readability, but way more important, it avoid an external modification of the value.